src/mesh/mesh_base.C

```diff
@@ -102,9 +102,27 @@ MeshBase::MeshBase (const MeshBase & other_mesh) :
   _elem_dims(other_mesh._elem_dims),
   _spatial_dimension(other_mesh._spatial_dimension),
   _default_ghosting(libmesh_make_unique<GhostPointNeighbors>(*this)),
-  _ghosting_functors(other_mesh._ghosting_functors),
   _point_locator_close_to_point_tol(other_mesh._point_locator_close_to_point_tol)
 {
+   for (const auto & gf : other_mesh._ghosting_functors )
+   {
+     std::shared_ptr<GhostingFunctor> clone_gf = gf->clone();
+     // Some subclass of GhostingFunctor does not implement clone
+     // GhostingFunctor will return nullptr by default.
+     // It is written for APIs upgrade
+     // Once moose is updated, we should come back
+     // and change this code
```

Comment:
This is extremely clever; good job!  I'd say let's throw in a libmesh_deprecated() in the else case, then we can leave it for a release or two and actually give everyone else a smooth upgrade path too.